### DISCUSSION TITLE
Older RTMP style codec identifier support for newer codecs

### DISCUSSION THREAD

**Discussion by mondain** [veovera/enhanced-rtmp#35] [Ideas]
I'd like to propose, for the sake of compatibility and in the spirit of the original codec identification scheme that the following modifications and additions be made to the enhanced specification. In short, that the following byte identifiers be used:

**Audio codec / sound format**

* MP2 - `0x0c`
* Opus - `0x0d`
* L16 - `0x0f` (reserved / device specific in spec)
* AC-3 - `0x10`
* EAC-3 - ` 0x11`
* FLAC - `0x12`

**Video codec**

* VP8 - `0x08`
* VP9 - `0x09`
* MP1 - `0x0b`
* HEVC - `0x0c`
* AV1 - `0x0d`

These changes should in no way preclude the implementation nor use of enhanced RTMP specified functionality.

Lastly, I am not aware of any devices or implementations on the market using any of these identifiers other than `0x0c` for **HEVC**, for which there are several available.

**Comment by zenomt** [comment]
i don't think this is needed, and would discourage the adoption and implementation of E-RTMP. code changes are already needed to support the new codecs and any codec-specific framing and special messages, so why not just do it the Enhanced way to begin with? extracting the codec ID the Enhanced way is literally just a few lines of code that should be factored to one place anyway, like: https://github.com/zenomt/rtmfp-cpp/blob/73fd63e6f3ae44de3554a1aa2c7b391fb7c560fe/src/TCMessage.cpp#L158-L176

also, allocating all of the remaining "classic" code points removes future flexibility for additional extension, should it ever be needed (like how audio was Enhanced by taking one of the unallocated code points).

**Reply by veovera** [reply]
Agree 100% that this is not needed.

**Comment by mondain** [comment]
Consider it pulled from ideas then; I understand the points made.
